### PR TITLE
plugins loading + preset plugin fetching implementation

### PR DESCRIPTION
**Pull request by lizlouise1335**
New plugins directory and init to handle loading and fetching of plugins
Change to dataprofiler/init to load any existing plugins at start
change to profiler_options to look for presets in plugins folder

### CODE REVIEW

File: dataprofiler/plugins/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+def plugin_decorator(typ, name):
+    def __inner_factory_function(fn):
+        global plugins_dict
+        plugins_dict[typ][name] = fn
+        return fn
+
+    return __inner_factory_function
+
+
+def loadPlugins():
```
Comment (JGSweets) [inline dataprofiler/plugins/__init__.py:18]: let's add some descriptions here to the functions

File: dataprofiler/plugins/__init__.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    plugin_path = os.path.dirname(os.path.abspath(__file__))
```
Comment (JGSweets) [inline dataprofiler/plugins/__init__.py:19]: smart

File: dataprofiler/plugins/__init__.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            for filename in os.listdir(option_path):
+                spec = importlib.util.spec_from_file_location(
+                    filename, os.path.join(option_path, filename)
+                )
+                if spec is not None:
+                    module = importlib.util.module_from_spec(spec)
+                    spec.loader.exec_module(module)
+
+
+def getPlugins(typ):
```
Comment (JGSweets) [inline dataprofiler/plugins/__init__.py:32]: descript here too

**Comment by JGSweets** [comment]
The raise error is not here bc it was not in dev yet. it is here: #909

**Comment by JGSweets** [comment]
Just need to add tests and then can get to a merge state.

File: dataprofiler/plugins/__init__.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    """
+    Fetches a dictionary of plugins of a certain type
+
+    :param typ: Broader classification/type of a plugin
+    :return: dict
+    """
+    # from . import plugins_dict
```
Comment (JGSweets) [inline dataprofiler/plugins/__init__.py:34]: remove this.

**Review by JGSweets — COMMENTED** [review]
Last suggestion would be to use snake_case for our func namings.

File: dataprofiler/plugins/__init__.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import importlib
+import os
+
+from dataprofiler.plugins.decorators import plugin_decorator, plugins_dict
```
Comment (JGSweets) [inline dataprofiler/plugins/__init__.py:4]: inside we use relative:
`from .decorators import plugin_decorator, plugins_dict`

**Review by taylorfturner — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]

**Review by tyfarnan — APPROVED** [review]

**Pull request merged**